### PR TITLE
Expose current capture device's torch availability

### PR DESCRIPTION
**Pull request by hvsw**
### Description of changes:
Expose current capture device's torch availability. Currently, to check current device torch availability, we need to query all video devices get from a `DicoverySession` then using `CameraCaptureSource.device.label` to match its name. This is bad because forces clients to know the internal workings of the SDK, so the client code can break at any time the SDK internal workings gets changed.

I'm adding this to the SDK for 2 main reasons:
- `captureDevice` is private so clients can’t extend it, and 
- `DefaultCameraCaptureSource` is not open so clients can’t subclass it. 

Currently, the SDK only exposes the ability to toggle the torch, and the sample app will fail since there's no torch available for the front-facing camera for example. It'd be good if someone implemented the usage of this new property in the sample apps.

|Option to turn it on for front camera|Fail to turn on the torch for front camera|
-|-
|image|image|

### CODE REVIEW

**Comment by hokyungh** [comment]
Thanks for opening, do you mind updating the branch to our latest development?

File: AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/capture/DefaultCameraCaptureSource.swift [hunk 1]
```diff
@@ -12,7 +12,16 @@ import UIKit
 
 @objcMembers public class DefaultCameraCaptureSource: NSObject, CameraCaptureSource {
     public var videoContentHint: VideoContentHint = .motion
+    
+    /// Expose current capture device's torch availability
+    public var isTorchAvailable: Bool {
```
Comment (hensmi-amazon) [inline AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/capture/DefaultCameraCaptureSource.swift:17]: Nit: Can we move this down to the other getters? i.e. `torchEnabled`.

Comment (hensmi-amazon) [inline AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/capture/DefaultCameraCaptureSource.swift:17]: Nit: `torchAvailable` to match file style

Comment (dylonChime) [inline AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/capture/DefaultCameraCaptureSource.swift:17]: You can replace `if let captureDevice = captureDevice, captureDevice.hasTorch, captureDevice.isTorchAvailable` on line 83 with `if let captureDevice = captureDevice, torchAvailable()`

**Comment by hvsw** [comment]
@hokyungh I originally branched off from `master`, can I just merge `development` into this branch?

**Review by linsang21 — APPROVED** [review]

**Review by dylonChime — APPROVED** [review]

**Comment by hvsw** [comment]
Would be great if someone could update the sample app to correctly reflect the torch availability.

**Pull request merged**